src/cli_plugin/install/zip.js

```diff
@@ -1,98 +1,120 @@
-import _ from 'lodash';
-import DecompressZip from '@bigfunger/decompress-zip';
-
-const SYMBOLIC_LINK = 'SymbolicLink';
+import yauzl from 'yauzl';
+import path from 'path';
+import mkdirp from 'mkdirp';
+import { createWriteStream } from 'fs';
+import { get } from 'lodash';
 
 /**
- * Creates a filter function to be consumed by extractFiles that filters by
- *  an array of files
- * @param {array} files - an array of full file paths to extract. Should match
- *   exactly a value from listFiles
+ * Returns an array of package objects. There will be one for each of
+ *  package.json files in the archive
+ *
+ * @param {string} archive - path to plugin archive zip file
  */
-function extractFilterFromFiles(files) {
-  const filterFiles = files.map((file) => file.replace(/\\/g, '/'));
-  return function filterByFiles(file) {
-    if (file.type === SYMBOLIC_LINK) return false;
-
-    const path = file.path.replace(/\\/g, '/');
-    return _.includes(filterFiles, path);
-  };
-}
 
-/**
- * Creates a filter function to be consumed by extractFiles that filters by
- *  an array of root paths
- * @param {array} paths - an array of root paths from the archive. All files and
- *   folders will be extracted recursively using these paths as roots.
- */
-function extractFilterFromPaths(paths) {
-  return function filterByRootPath(file) {
-    if (file.type === SYMBOLIC_LINK) return false;
+export function analyzeArchive(archive) {
+  const plugins = [];
+  const regExp = new RegExp('(kibana[\\\\/][^\\\\/]+)[\\\\/]package\.json', 'i');
 
-    return paths.some(path => {
-      const regex = new RegExp(`${path}($|/)`, 'i');
-      return file.parent.match(regex);
-    });
-  };
-}
+  return new Promise ((resolve, reject) => {
+    yauzl.open(archive, { lazyEntries: true }, function (err, zipfile) {
+      if (err) {
```

Comment:
Does `yauzl` document possible `err` objects?  If so, we should include some of the information from the error object in the log message rather than swallowing it, like the message/stack trace.  Things like permission errors during install are common enough that we should provide people with some sort of specific feedback where possible.